### PR TITLE
chore: update to svelte5

### LINKED ISSUE #7824: update to svelte v5
### Is your enhancement related to a problem? Please describe

runes, snippets, events

### Describe the solution you'd like

bump to svelte5 as there is no new feature being introduced

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### LINKED ISSUE #7837: svelte: replace event handler on: by callback functions
### Is your enhancement related to a problem? Please describe

We should replace all usage of custom event handlers using the on: approach by using specific callback methods 

For example instead of `on:click` to have `onclick` callback property

### Describe the solution you'd like

replace all custom event handlers by specific callback methods 

### Describe alternatives you've considered

_No response_

### Additional context

Event handlers are becoming deprecated so better to not use them

### PR DESCRIPTION
**Pull request by benoitf**
### What does this PR do?
Update to svelte v5 RC

update also storybook svelte to the experimental version to be able to have something working on storybook

### What issues does this PR fix or reference?

fixes https://github.com/containers/podman-desktop/issues/7824
fixes https://github.com/containers/podman-desktop/issues/7837

### How to test this PR?

- [ ] Tests are covering the bug fix or the new feature

### CODE REVIEW

File: packages/renderer/src/lib/dashboard/NotificationCardItem.svelte [hunk 1]
```diff
@@ -22,12 +22,12 @@ async function removeNotification(id: number) {
     </div>
     <div class="flex flex-col text-sm space-y-2">
       <div class="font-bold" aria-label="Notification title">{notification.title}</div>
-      <div aria-label="Notification description"><Markdown>{notification.body ?? ''}</Markdown></div>
+      <div aria-label="Notification description"><Markdown markdown="{notification.body ?? ''}" /></div>
```
Comment (axel7083) [inline packages/renderer/src/lib/dashboard/NotificationCardItem.svelte:25]: Why does the markdown is changed ?

File: packages/renderer/src/lib/deployments/DeploymentsList.spec.ts [hunk 2]
```diff
@@ -22,7 +22,10 @@ import '@testing-library/jest-dom/vitest';
 
 import type { V1Deployment } from '@kubernetes/client-node';
 import { render, screen, within } from '@testing-library/svelte';
+/* eslint-disable import/no-duplicates */
+import { tick } from 'svelte';
 import { get } from 'svelte/store';
+/* eslint-enable import/no-duplicates */
```
Comment (axel7083) [inline packages/renderer/src/lib/deployments/DeploymentsList.spec.ts:28]: ?? vitest is only imported once

File: packages/renderer/src/lib/extensions/ExtensionDetailsReadme.spec.ts [hunk 3]
```diff
@@ -19,15 +19,19 @@
...
-  while (result.component.$$.ctx[0] === undefined) {
-    await new Promise(resolve => setTimeout(resolve, 100));
-  }
+  render(ExtensionDetailsReadme, { ...customProperties });
+  await tick();
+  await tick();
+  await tick();
+  await tick();
+  await tick();
+  await tick();
```
Comment (axel7083) [inline packages/renderer/src/lib/extensions/ExtensionDetailsReadme.spec.ts:34]: why

File: packages/renderer/src/lib/featured/FeaturedExtensionDownload.spec.ts [hunk 4]
```diff
@@ -59,7 +59,7 @@ test('Expect that the install button is hidden if extension is not installable',
 });
 
 test('Expect that we can see the button and click on the install', async () => {
-  const featuredExtension: FeaturedExtension = {
+  let featuredExtension: FeaturedExtension = {
```
Comment (axel7083) [inline packages/renderer/src/lib/featured/FeaturedExtensionDownload.spec.ts:62]:
```suggestion
  const featuredExtension: FeaturedExtension = {
```

I don't see any write after the first initialization?

File: packages/renderer/src/lib/pod/KubernetesTerminal.spec.ts [hunk 5]
```diff
@@ -19,7 +19,10 @@
 import '@testing-library/jest-dom/vitest';
 
 import { render, waitFor } from '@testing-library/svelte';
+/* eslint-disable import/no-duplicates */
+import { tick } from 'svelte';
 import { get } from 'svelte/store';
+/* eslint-enable import/no-duplicates */
```
Comment (axel7083) [inline packages/renderer/src/lib/pod/KubernetesTerminal.spec.ts:25]: No duplicates ?

File: packages/renderer/src/lib/pod/PodsList.spec.ts [hunk 6]
```diff
@@ -22,7 +22,10 @@ import '@testing-library/jest-dom/vitest';
 
 import { fireEvent, render, screen } from '@testing-library/svelte';
 import userEvent from '@testing-library/user-event';
+/* eslint-disable import/no-duplicates */
```
Comment (axel7083) [inline packages/renderer/src/lib/pod/PodsList.spec.ts:25]: no duplicates ?

File: packages/renderer/src/lib/pod/PodsList.spec.ts [hunk 7]
```diff
@@ -22,7 +22,10 @@ import '@testing-library/jest-dom/vitest';
...
+import { tick } from 'svelte';
 import { get } from 'svelte/store';
+/* eslint-enable import/no-duplicates */
```
Comment (axel7083) [inline packages/renderer/src/lib/pod/PodsList.spec.ts:28]: no duplicates ?

File: packages/renderer/src/lib/service/ServicesList.spec.ts [hunk 8]
```diff
@@ -22,7 +22,10 @@ import '@testing-library/jest-dom/vitest';
 
 import type { V1Service } from '@kubernetes/client-node';
 import { render, screen } from '@testing-library/svelte';
+/* eslint-disable import/no-duplicates */
+import { tick } from 'svelte';
 import { get } from 'svelte/store';
+/* eslint-enable import/no-duplicates */
```
Comment (axel7083) [inline packages/renderer/src/lib/service/ServicesList.spec.ts:28]: no duplicates ?

File: packages/renderer/src/lib/service/ServicesList.spec.ts [hunk 9]
```diff
@@ -22,7 +22,10 @@ import '@testing-library/jest-dom/vitest';
 
 import type { V1Service } from '@kubernetes/client-node';
 import { render, screen } from '@testing-library/svelte';
+/* eslint-disable import/no-duplicates */
```
Comment (axel7083) [inline packages/renderer/src/lib/service/ServicesList.spec.ts:25]: no duplicates ?

File: packages/renderer/src/lib/volume/VolumesList.spec.ts [hunk 10]
```diff
@@ -22,7 +22,10 @@ import '@testing-library/jest-dom/vitest';
 
 import { render, screen } from '@testing-library/svelte';
 import userEvent from '@testing-library/user-event';
+/* eslint-disable import/no-duplicates */
+import { tick } from 'svelte';
 import { get } from 'svelte/store';
+/* eslint-enable import/no-duplicates */
```
Comment (axel7083) [inline packages/renderer/src/lib/volume/VolumesList.spec.ts:28]: no duplicates ?

File: packages/renderer/src/lib/volume/VolumesList.spec.ts [hunk 11]
```diff
@@ -22,7 +22,10 @@ import '@testing-library/jest-dom/vitest';
 
 import { render, screen } from '@testing-library/svelte';
 import userEvent from '@testing-library/user-event';
+/* eslint-disable import/no-duplicates */
```
Comment (axel7083) [inline packages/renderer/src/lib/volume/VolumesList.spec.ts:25]: no duplicates ?

File: packages/ui/src/lib/layouts/DetailsPage.svelte [hunk 12]
```diff
@@ -13,19 +13,18 @@ export let breadcrumbTitle: string | undefined = '';
 
 const dispatchClose = createEventDispatcher<{ close: undefined }>();
 
-export function close(): void {
+export let onclose: () => void = () => {
```
Comment (axel7083) [inline packages/ui/src/lib/layouts/DetailsPage.svelte:16]: onClose ?

File: packages/ui/src/lib/link/Link.svelte [hunk 13]
```diff
@@ -10,6 +10,9 @@ export let icon: any = undefined;
 let iconType: string | undefined = undefined;
 
 const dispatch = createEventDispatcher<{ click: undefined }>();
+export let onclick: () => void = () => {
```
Comment (axel7083) [inline packages/ui/src/lib/link/Link.svelte:13]: onClick ?

File: packages/ui/src/lib/modal/Modal.svelte [hunk 14]
```diff
@@ -4,15 +4,17 @@ import { createEventDispatcher, onDestroy } from 'svelte';
 import { tabWithinParent } from '../utils/dialog-utils';
 
 const dispatch = createEventDispatcher();
-const close = (): boolean => dispatch('close');
 
 let modal: HTMLDivElement;
 export let name = '';
 export let top: boolean = false;
+export let onclose: () => void = () => {
```
Comment (axel7083) [inline packages/ui/src/lib/modal/Modal.svelte:11]: onClose ?

**Review by axel7083 — COMMENTED** [review]
Great job at migrating our codebase to svelte5, left some comments 👍

File: packages/renderer/src/lib/dashboard/NotificationCardItem.svelte (hunk 1 above)
Comment (benoitf) [reply packages/renderer/src/lib/dashboard/NotificationCardItem.svelte:25]: it looks like with svelte5 I had at some point the content not being displayed using the other syntax but it was also not 100%

File: packages/renderer/src/lib/pod/KubernetesTerminal.spec.ts (hunk 5 above)
Comment (benoitf) [reply packages/renderer/src/lib/pod/KubernetesTerminal.spec.ts:25]: the no-duplicates rule is for svelte imports

by default the plug-in by seeing 
```
import { tick } from 'svelte';
import { get } from 'svelte/store';
```

will migrate to
```
import {get, tick} from 'svelte'
```

due so some exposures and it'll fail this is why all imports when importing both from the core and the store need to be in a no-duplicates block 

it's not related to svelte5, we already do this and there is a bug referenced somewhere for svelte/the plugin

File: packages/renderer/src/lib/deployments/DeploymentsList.spec.ts (hunk 2 above)
Comment (benoitf) [reply packages/renderer/src/lib/deployments/DeploymentsList.spec.ts:28]: it's not for vitest, it's for the block between disable/enable for svelte

File: packages/ui/src/lib/modal/Modal.svelte (hunk 14 above)
Comment (benoitf) [reply packages/ui/src/lib/modal/Modal.svelte:11]: it seems svelte is using onclick, onclose (vs on:click) (not onClose)

File: packages/renderer/src/lib/extensions/ExtensionDetailsReadme.spec.ts (hunk 3 above)
Comment (benoitf) [reply packages/renderer/src/lib/extensions/ExtensionDetailsReadme.spec.ts:34]: onMount is asynchronous so by default after a call to the render method the onMount method may not be completed

before we were monitoring the $$ctx object to see if some variables were defined (and $$ctx is no longer there with svelte5) and the recommended way is to tick

for this specific component, as there is a lot of different context switching in the onMount method I had to tick the number of times to have the component being ready

File: packages/renderer/src/lib/pod/PodsList.spec.ts (hunk 6 above)
Comment (benoitf) [reply packages/renderer/src/lib/pod/PodsList.spec.ts:25]: I commented before it's for svelte,  svelte/store imports

File: packages/renderer/src/lib/pod/KubernetesTerminal.spec.ts (hunk 5 above)
Comment (benoitf) [reply packages/renderer/src/lib/pod/KubernetesTerminal.spec.ts:25]: the original issue https://github.com/import-js/eslint-plugin-import/issues/1479

Comment (axel7083) [reply packages/renderer/src/lib/pod/KubernetesTerminal.spec.ts:25]: Thanks for the details

File: packages/ui/src/lib/modal/Modal.svelte (hunk 14 above)
Comment (axel7083) [reply packages/ui/src/lib/modal/Modal.svelte:11]: Okey weird camel rules but okey

**Review by axel7083 — APPROVED** [review]
LGTM, tested and seems that what I covered is working

File: packages/renderer/package.json [hunk 15]
```diff
@@ -34,7 +36,7 @@
     "monaco-editor": "^0.50.0",
     "postcss": "^8.4.38",
     "postcss-load-config": "^6.0.1",
-    "svelte": "4.2.18",
+    "svelte": "5.0.0-next.164",
```
Comment (axel7083) [inline packages/renderer/package.json:39]: [svelte@5.0.0-next.168](https://github.com/sveltejs/svelte/releases/tag/svelte%405.0.0-next.168)

Comment (benoitf) [reply packages/renderer/package.json:39]: the issue is that for example 166 was not working so I prefer to have something first working and then we do the update

Comment (axel7083) [reply packages/renderer/package.json:39]: Okey so https://github.com/containers/podman-desktop/issues/7824 should stay open after the merge of this PR

Comment (benoitf) [reply packages/renderer/package.json:39]: @axel7083 not sure to follow why it should stay open. I don't see the difference if it's 164 or 166 or 167
it comes to dependabot upgrades after it has been merged.

Comment (axel7083) [reply packages/renderer/package.json:39]: You said 166 was not compatible with current, so fix are still needed to be compliant with v5 ?

Comment (benoitf) [reply packages/renderer/package.json:39]: but it goes into the usual 'update libs to their latest version' workflow managed by dependabot

Comment (axel7083) [reply packages/renderer/package.json:39]: Okey

File: package.json [hunk 16]
```diff
@@ -82,7 +82,7 @@
     "lint:clean": "rimraf .eslintcache",
     "lint:fix": "node --max-old-space-size=6144 node_modules/eslint/bin/eslint.js --cache . --fix --ext js,ts,tsx,svelte",
     "lint:check": "node --max-old-space-size=6144 node_modules/eslint/bin/eslint.js --cache . --ext js,ts,tsx,svelte",
-    "svelte:check": "svelte-check --ignore \"packages/ui/dist\"",
+    "svelte:check": "svelte-check --ignore \"packages/ui/dist,packages/renderer/src/lib/ui/DetailsPage.svelte,packages/renderer/src/lib/ui/EngineFormPage.svelte,packages/renderer/src/lib/ui/FormPage.svelte,storybook\"",
```
Comment (jeffmaury) [inline package.json:85]: What's the rationale for these exceptions ?

Comment (benoitf) [reply package.json:85]: the slots elements are 'children' now but it should be using snippets
And I didn't found any way of being able do declare 'children' in the parent components

**Pull request merged**